### ISSUE TITLE
Memory leak when to_json is used on an array column.

### ISSUE THREAD

**Issue by jorgecarleitao** [pandas-dev/pandas#26348]
#### Code Sample, a copy-pastable example if possible

```python
import resource

import pandas as pd

# some random data, some of them as array columns
batches = 5000
df = pd.DataFrame({
    't': [pd.np.array(range(0, 180 * 60, 5))] * batches,
})


# read the data above and convert it to json (e.g. the backend of a restful API)
for i in range(100):
    # comment any of the 2 lines for the leak to vanish.
    df['t'] = df['t'].apply(lambda x: pd.np.array(list(x)))  # see comment below
    df['t'].to_json()
    print(resource.getrusage(resource.RUSAGE_SELF).ru_maxrss)
```
#### Problem description

The code above gives the following result (only works on linux/Mac)

```
254384
337780
422984
508720
593996
679336
762564
848036
933768
1019040
1104384
1187612
1273084
1358552
1444088
1529432
1612660
1698128
1783600
1869136
1954244
2037708
2122980
...
```

i.e. the memory peak is increasing, aka a memory leak.

The line that does a seemmesly stupid operation is important because it is the equivalent of reading from a Parquet file using pyarrow. See [issue 5302](https://issues.apache.org/jira/projects/ARROW/issues/ARROW-5302) for a script that reproduces this within pyarrow.

Regardless of how "dumb" the code above is, it demonstrates a memory leak, which should not happen in Python. The code on the issue above demonstrates an important use-case where this leak emerges.

#### Expected Output

The values above should not be unbounded.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.4.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-1075-aws
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.2
pytest: 4.0.2
pip: 9.0.1
setuptools: 38.4.0
Cython: 0.27.3
numpy: 1.14.0
scipy: 1.0.0
pyarrow: 0.13.0
xarray: None
IPython: 6.2.1
sphinx: 1.6.6
patsy: 0.5.0
dateutil: 2.6.1
pytz: 2017.3
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.4
feather: None
matplotlib: 2.1.2
openpyxl: 2.4.10
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.2
lxml.etree: 4.1.1
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.2.1
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: 0.2.1
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

**Comment by gfyoung** [comment]
@jorgecarleitao : Thanks for reporting this!  Can you try running this code against `master` ?  I tried it there, and I don't see any noticeable leak.

**Comment by jorgecarleitao** [comment]
Cannot comply: the machine I have for this has no access to the internet. I did used the latest release with on a clean fresh venv (`Python 3.6.4 :: Anaconda, Inc.`), and the error is there:

```
numpy==1.16.3
pandas==0.24.2
python-dateutil==2.8.0
pytz==2019.1
six==1.12.0
```

**Comment by gfyoung** [comment]
@jorgecarleitao : If your machine has no access to the internet, how did you setup your Python environment in the first place?  I'm trying to see whether or not you can perform a similar method to install `master` on this machine.

**Comment by jorgecarleitao** [comment]
Ok, so, I have ran it on a Mac, 3.7, on two venvs each with a specific version of pandas:

1. python -m venv venv1 && pip install pandas
2. python -m venv venv2 && pip install cython && pip install git+https://github.com/pandas-dev/pandas.git@bb0376c209d97e49b9c94038926ee6

The first 1 leaks, the second does not.

So, @gfyoung , yes, master does not leak. This issue was solved between 0.14.2 and bb0376c209d97e49b9c94038926ee6. :)
